src/Core/src/HttpClient/RetryResponse.php

```diff
@@ -0,0 +1,213 @@
+        if (!$this->initialized) {
+            $this->initialize();
+        }
+
+        return $this->inner->toArray($throw);
+    }
+
+    public function cancel(): void
+    {
+        $this->initialized = true;
+        $this->inner->cancel();
+    }
+
+    public function getInfo(string $type = null)
+    {
+        if (!$this->initialized) {
+            $this->initialize();
+        }
+
+        return $this->inner->getInfo($type);
+    }
+
+    public static function stream(iterable $responses, float $timeout = null): \Generator
+    {
+        $wrappedResponses = [];
+        $map = new \SplObjectStorage();
+        $client = null;
+
+        foreach ($responses as $r) {
+            if (!$r instanceof self) {
+                throw new \TypeError(sprintf('"%s::stream()" expects parameter 1 to be an iterable of "%s" objects, "%s" given.', static::class, self::class, get_debug_type($r)));
+            }
+
+            if (null === $client) {
+                $client = $r->client;
+            } elseif ($r->client !== $client) {
+                throw new TransportException(sprintf('Cannot stream "%s" objects with many clients.', self::class));
+            }
+        }
+
+        if (!$client) {
+            return;
+        }
+
+        $end = null;
+        if (null !== $timeout) {
+            $end = \microtime(true) + $timeout;
+        }
+
+        while (true) {
+            foreach ($responses as $r) {
+                $wrappedResponses[] = $r->inner;
+                $map[$r->inner] = $r;
+            }
+            $subTimeout = null;
+            if (null !== $end && ($subTimeout = $end - \microtime(true)) <= 0) {
+                foreach ($map as $response) {
+                    yield $response => new ErrorChunk(0, sprintf('Idle timeout reached for "%s".', $response->getInfo('url')));
+                }
+
+                break;
+            }
+
+            foreach ($client->stream($wrappedResponses, $subTimeout) as $response => $chunk) {
+                /** @var self $r */
+                $r = $map[$response];
+                if (!$chunk->isTimeout() && $chunk->isFirst()) {
+                    if ($r->handleRetry()) {
+                        continue 2;
+                    }
+                }
+                yield $r => $chunk;
+            }
+
+            break;
+        }
+    }
+
+    private function initialize(): void
+    {
+        while (!$this->initialized) {
+            $this->handleRetry();
+        }
+    }
+
+    /**
+     * @return bool return true when the request have been retried
+     *
+     * @throws TransportExceptionInterface
+     */
+    private function handleRetry(): bool
+    {
+        if ($this->initialized) {
+            return false;
+        }
+
+        $handle = function (string $message, array $context): bool {
+            if (++$this->tryCount <= $this->maxTryCount) {
+                $this->logger->info($message . ' Retry the request {attempt}/{maxAttempts}.', $context + [
```

Comment:
```suggestion
                $this->logger->info($message . ' Retry the request {attempt} of {maxAttempts}.', $context + [
```

It seams weird if the retry request 2 thirds (2/3)

Reply:
fixed